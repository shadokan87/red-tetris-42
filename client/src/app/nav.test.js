import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Navbar from "./nav";
import React from "react";
import { useRouter } from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));

beforeAll(() => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
});

test("renders navbar", () => {
  const mockStore = configureMockStore();
  const store = mockStore({
    session: {
      user: null,
      axiosConfig: null,
      token: "mocked_token",
    },
    lobby: {
      room: null,
    },
  });

  const navItems = [
    { label: "Room", pathname: "/room", disabled: true },
    { label: "Lobby", pathname: "/lobby", disabled: false },
  ];

  const { asFragment } = render(
    <Provider store={store}>
      <Navbar navItems={navItems} />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
