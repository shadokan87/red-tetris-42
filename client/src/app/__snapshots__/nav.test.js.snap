// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar 1`] = `
<DocumentFragment>
  <div
    class="ant-flex css-dev-only-do-not-override-xu9wm8 ant-flex-justify-space-between"
  >
    <div
      class="ant-flex css-dev-only-do-not-override-xu9wm8"
      classlabel="navbar"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-xu9wm8 ant-btn-link"
        disabled=""
        type="button"
      >
        <span>
          Room
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-xu9wm8 ant-btn-link"
        type="button"
      >
        <span>
          Lobby
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
